src/main.js

```diff
@@ -18,36 +19,77 @@ const filters = generateFilter(filmCards);
 const movieSortByRating = filmCards.slice().sort((a,b)=> b.filmInfo.rating - a.filmInfo.rating);
 const movieSortByComments = filmCards.slice().sort((a,b)=> b.comments.length - a.comments.length);
 
-const render = (container, template, place) => {
-  container.insertAdjacentHTML(place, template);
-};
+const bodyElement = document.querySelector('body');
 const headerElement = document.querySelector('.header');
 const siteMainElement = document.querySelector('.main');
 const footerElement = document.querySelector('.footer');
 
-render(headerElement, createHeaderProfileTemplate(filters.alreadyWatched), 'beforeend');
-render(siteMainElement,createSiteMenuTemplate(filters), 'beforeend');
-render(siteMainElement,createSortFilmsTemplate(), 'beforeend');
-render(siteMainElement,createContentAreaTemplate(), 'beforeend');
-render(footerElement,createMovieCountTemplate(filmCards), 'beforeend');
-render(footerElement,createFilmPopupTemplate(filmCards[0]), 'afterend');
+const renderFilmCard = (filmCardsContainer, filmCard) => {
+  const filmCardComponent = new FilmCardView(filmCard);
+  const filmPopupComponent = new FilmPopupView(filmCard);
+
+  const closePopup = () => {
+    bodyElement.removeChild(filmPopupComponent.getElement());
+    filmPopupComponent.getElement().querySelector('.film-details__close-btn').removeEventListener('click', closePopup);
+  };
+
+  const onEscKeyDown = (evt) => {
+    if (evt.key === 'Escape' || evt.key === 'Esc') {
```

Comment:
```suggestion
    if (evt.key === KeyCode.ESPACE) {
```
лучше на это создать енам